### PR TITLE
expose variableEditorOpen to set a default value

### PR DESCRIPTION
**Pull request by christieb7007**
https://github.com/graphql/graphiql/issues/741

### CODE REVIEW

**Comment by christieb7007** [comment]
can this be merged?

**Comment by acao** [comment]
@christieb7007 looks good to me, other than the unneeded diff to `package.json`. can you remove that change from the PR?

**Comment by christieb7007** [comment]
I've updated the PR so that the `variableEditorOpen` prop is evaluated in the `render()` method as well as update tests/README/example

**Comment by christieb7007** [comment]
Apologies for the delay.
The functionality I was thinking of was actually just the `defaultVariableEditorOpen` - maybe the controlled component part can be in a separate PR.

File: packages/graphiql-examples/cdn/index.html [hunk 1]
```diff
@@ -143,6 +143,7 @@
           operationName: parameters.operationName,
           onEditQuery: onEditQuery,
           onEditVariables: onEditVariables,
+          variableEditorOpen: true,
```
Comment (benjie) [inline packages/graphiql-examples/cdn/index.html]:
```suggestion
          defaultVariableEditorOpen: true,
```

File: packages/graphiql/src/components/GraphiQL.js [hunk 2]
```diff
@@ -120,6 +121,9 @@ export class GraphiQL extends React.Component {
       docExplorerOpen = this._storage.get('docExplorerOpen') === 'true';
     }
 
+    // initial variable editor pane open
+    const variableEditorOpen = props.defaultVariableEditorOpen !== undefined ? props.defaultVariableEditorOpen : Boolean(variables);
```
Comment (benjie) [inline packages/graphiql/src/components/GraphiQL.js]: (Note to self: comparing against `undefined` should be safe in modern browsers; I think this is okay.)

File: packages/graphiql-examples/cdn/index.html (hunk 1 above)
Comment (christieb7007) [reply packages/graphiql-examples/cdn/index.html]: fixed

**Pull request merged**